supervise.py

```diff
@@ -267,9 +267,13 @@ def stop(self):
         self._signal("d")
 
     def _signal(self, signal):
-        f = open(self._control,"w+")
-        f.write(signal)
-        f.close()
+        try:
+            f = open(self._control,"w+")
+            f.write(signal)
+        except IOError as e:
+            raise ServiceControlError("Unable to write service control file '{0}'".format(e.filename))
+        else:
```

Comment:
This is kind of sketchy because of the write fails, we still want to close the file.

Probably initialize f = None outside of the try block and do a conditional close in a finally: